rustls/src/enums.rs

```diff
@@ -565,6 +565,20 @@ enum_builder! {
     }
 }
 
+enum_builder! {
```

Comment:
@djc I'm picking up this work again and wanted to check in. I think reworking the commits so that updates like this one are introduced alongside the changes that require them as you requested makes sense :+1: 

How do you feel about something like breaking up offering ECH and checking for ECH offer acceptance into two commits if it means putting a `todo!` or two in the first commit to stub out things that will be done in the second? It might be helpful to try and reduce the individual commit size but comes at the cost of making the overall "feature" not contained in a single commit. WDYT?

Reply:
I'm open to it if you think it makes the PR easier to review overall.

Reply:
Thanks, that's useful to know. I'll see what I can come up with trying to balance commit size & avoiding too much churn.

Reply:
I've reworked based on the strategy I pitched above. I think it's the best balance of reviewability and keeping commits a manageable size. I think the way the protocol is implemented in a few pieces is also a very useful aide for others looking at implementing the RFC because it's _very_ non-linear compared to this commit stack.